src/openms/source/FORMAT/MSPGenericFile.cpp

```diff
@@ -131,20 +132,42 @@ namespace OpenMS
       {
         spectrum.setRT(std::stod(m[1]));
       }
-      // Specific case of NIST's exported msp
+      //CAS# NIST#
       else if (boost::regex_search(line, m, re_cas_nist))
       {
         // OPENMS_LOG_DEBUG << "CAS#: " << m[1] << "; NIST#: " << m[2] << "\n";
         spectrum.setMetaValue(String("CAS#"), String(m[1]));
         spectrum.setMetaValue(String("NIST#"), String(m[2]));
       }
-      // Other metadata
+      // Other metadata, needs to be last, matches everything
```

Comment:
Do a special case for PRECURSORMZ, such that you can parse to double and set precursor right away?

Reply:
What do you think about special cases for meta values required for MetaboliteSpectralMatching? This would be an alternative to renaming the meta values later.

Reply:
Yes true, maybe better. I think some additional regex will not be the bottleneck and you will save the metavalue renaming.